sql/persisted_variable.cc

```diff
@@ -332,17 +334,19 @@ bool Persisted_variables_cache::set_variable(THD *thd, set_var *setvar) {
     uint dummy_err;
     String bool_str;
     if (setvar->value) {
-      res = setvar->value->val_str(&str);
+      setvar->var->persist_only_update(thd, setvar, str.ptr());
```

Comment:
Here we pass str.ptr() to the function where we copy data to this buffer. But inside we do not know about how big the buffer is. Maybe it is better to pass String* instead of char* and construct the string in safe manner?

Reply:
Changed to pass a String instead of *char.